### ISSUE TITLE
Training label on RIS CNN in '06_train_super.py'

### ISSUE THREAD

**Issue by GGuuu** [khinthandarkyaw98/Optimization-of-Transmit-Beamforming-and-RIS-Coefficients-Using-Channel-Covariances-in-MISO-Downlink#1]
Hi!

I've been seen your journal recently, and I was impressed with your work.
However, I have some questions in the logic of training for RIS CNN in '06_train_super.py'

```
...
G_stacked_input = layers.Input(name='GStackedInput', shape=(G_stacked.shape[1:]), dtype=tf.float32)
Rg_stacked_input = layers.Input(name='RgStackedInput', shape=(Rg_stacked.shape[1:]), dtype=tf.float32)
phi_input = layers.Input(name='PhiInput', shape=(phi_train.shape[1:]), dtype=tf.float32)

G_branch = build_branch(G_stacked_input)
Rg_branch = build_branch(Rg_stacked_input)
phi_flat = layers.Flatten()(phi_input)

concatenated = layers.Concatenate()([G_branch, Rg_branch, phi_flat])
x = layers.BatchNormalization()(concatenated)

...

loss = layers.Lambda(loss_phi, 
                     dtype=tf.float32, 
                     output_shape=(1),
                     name="loss")([phi_input, phi_pred, N])
```

The problems are the 'concatenated' and 'loss'.
For `concatenated`, you are concatenating G, Rg and flattened `phi_input`.
And I found that `phi_input` is used as a label of the samples in `loss_phi()` which returns value to the variable of `loss`.

```
def loss_phi(temp):
    phi_true, phi_pred, N = temp
    N_casted = tf.cast(N, dtype=tf.float32)
    phi_diff = tf.subtract(phi_true, phi_pred)
    phi_diff_norm = tf.linalg.norm(phi_diff, ord='euclidean', axis=[2, 3])
    loss = (1.0 / tf.sqrt(N_casted)) * tf.reduce_sum(phi_diff_norm, axis=1)
    # Compute the mean loss across samples and users
    loss = tf.reduce_mean(loss)
    return loss
```
In the function `loss_phi()`, inputted `phi_input` goes to `phi_true`.

So in this training logic, I think you are using this label data (`phi_input`) for training RIS CNN.

If I have misunderstanding with this code, please let me know.

Thank you.

**Comment by khinthandarkyaw98** [comment]
Hello !!!

Thank you for your keen observation. In our original design, we included `phi_input` (optimal RIS coefficients from ADMM) as an input feature to guide the model, intending to leverage prior knowledge. However, as you noted, this risks data leakage since `phi_input `correlates with the training labels. 

We will try to revise and update the code in the future. 

We truly appreciate you for pointing this out.
